adapters/clients/client.go

```diff
@@ -62,13 +63,14 @@ func (c *retryClient) doWithCustomMarshaller(timeout time.Duration,
 func (c *retryClient) do(timeout time.Duration, req *http.Request, body []byte, resp interface{}, success func(code int) bool) (code int, err error) {
 	ctx, cancel := context.WithTimeout(req.Context(), timeout)
 	defer cancel()
+	req = req.WithContext(ctx)
 	try := func(ctx context.Context) (bool, error) {
 		if body != nil {
 			req.Body = io.NopCloser(bytes.NewReader(body))
 		}
 		res, err := c.client.Do(req)
 		if err != nil {
-			return ctx.Err() == nil, fmt.Errorf("connect: %w", err)
+			return false, fmt.Errorf("connect: %w", err)
```

Comment:
thought(non-blocking): Would it make sense here to improve the error message with maybe the URL or the Host we were trying to reach ? The idea would be to improve the weaviate logs when this get logged further down the line.

Reply:
No need because the URL is part of original error